### PR TITLE
hmwk18

### PR DESCRIPTION
**Pull request by fallonwilliams**
validate song is playing

### CODE REVIEW

**Comment by fallonwilliams** [comment]
merge

File: src/test/resources/TestNG.xml [hunk 1]
```diff
@@ -7,7 +7,11 @@
         <classes>
 
 
-            <class name="Hmwk18"></class>
+            <class name="Hmwk18">
+            <methods>
+                <include name=""/>
```
Comment (Berichicko) [inline src/test/resources/TestNG.xml:12]: Please add  `<include name="AddSongToPlaylist"/>`

For example

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >

<suite name="TestNG">
    <parameter name="baseURL" value="https://bbb.testpro.io" />
    <test name="Example Test" preserve-order="false">
        <classes>
            <class name="Hmwk18">
            <methods>
                <include name="AddSongToPlaylist"/>
            </methods>
            </class>
        </classes>
    </test>
</suite>
```

File: src/test/java/Hmwk18.java [hunk 2]
```diff
@@ -37,9 +38,14 @@ public void AddSongToPlaylist() throws InterruptedException {
...
 
+    public void ValidateSongIsPlaying(){
+        WebElement ValidateSongPlaying = driver.findElement(By.xpath("//button[@title='Click for a marvelous visualizer']"));
+        ValidateSongPlaying.click();
+        Assert.assertTrue(ValidateSongPlaying.isDisplayed());
+    }
     private void PlaySong() {
 
-        WebElement PlaySong = driver.findElement(By.xpath("//ul/li['playback]"));
+        WebElement PlaySong = driver.findElement(By.xpath("//span[data-testid='play-btn']"));
```
Comment (Berichicko) [inline src/test/java/Hmwk18.java:48]: Try like this:

 ```
private void PlaySong() {
        WebElement PlaySong = driver.findElement(By.xpath("//span[@data-testid='play-btn']"));
        Actions actions = new Actions(driver);
        actions.click(PlaySong).perform();

    }

```

image

File: src/test/java/BaseTest.java [hunk 3]
```diff
@@ -51,8 +53,9 @@ public void providePassword(String password) {
 
     }
 
+    @Contract(value = " -> new", pure = true)
```
Comment (Berichicko) [inline src/test/java/BaseTest.java:56]: Code refactoring BaseTest

For example

```
import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.testng.annotations.*;
import java.time.Duration;

public class BaseTest {

    String url;
    static WebDriver driver;
    @BeforeSuite
    public static void chromeConfigs() {

        if (System.getProperty("os.name").toLowerCase().contains("win")) {
            System.setProperty("webdriver.chrome.driver", "chromedriver.exe");
        }
    }
    @BeforeMethod
    public void launchBrowser() {
        ChromeOptions chrome_Profile = new ChromeOptions();
        chrome_Profile.addArguments("--disable-notifications");
        driver = new ChromeDriver(chrome_Profile);
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
        driver.manage().window().maximize();
        url = "https://bbb.testpro.io/#!/home";
        driver.get(url);
    }


    public void clickSubmitBtn() {
        WebElement submitButton = driver.findElement(By.cssSelector("[type='submit']"));
        submitButton.click();
    }

    public void provideEmail(String email) {
        WebElement emailField = driver.findElement(By.cssSelector("[type='email']"));
        emailField.click();
        emailField.sendKeys(email);
    }

    public void providePassword(String password) {
        WebElement passwordField = driver.findElement(By.cssSelector("[type='password']"));
        passwordField.click();
        passwordField.sendKeys(password);

    }

    @Contract(value = " -> new", pure = true)
    @DataProvider(name="invalidCredentials")
    public static Object[] @NotNull [] getCredentials(){

        return new Object[][] {
                {"invalid@class.com", "invalidPass"},
                {"d@class.com", ""},
                {"", ""}
        };
    }

    @AfterMethod
    public void closeBrowser() {
          driver.quit();
    }

}
```

File: src/test/java/Hmwk18.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        songList.click();
+
+    }
+
+    private void clickViewAllBtn() {
+        WebElement viewAllBtn = driver.findElement(By.xpath("//button[@data-testid='home-view-all-recently-played-btn']"));
+        viewAllBtn.click();
+    }
+}
+
```
Comment (Berichicko) [inline src/test/java/Hmwk18.java:87]: The code is running but please fix it

image


I have corrected your code a bit, here is an example

```
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;
import org.testng.Assert;
import org.testng.annotations.Test;

public class Hmwk18 extends BaseTest{


    @Test
    public void AddSongToPlaylist() throws InterruptedException {
        provideEmail("seamugg@yahoo.com");
        providePassword("te$t$tudent");
        Thread.sleep(3000);
        clickSubmitBtn();
        Thread.sleep(3000);
        clickViewAllBtn();
        Thread.sleep(3000);
        String getSongTitle = getSongTitleTxt();
        clickSongFromList();
        clickAddToPlaylist();
        addToSuperPlaylist();
        PlaySong();
        ValidateSongIsPlaying();


        //Verify the song
        String songFromSuperPlaylist = getSongTitleFromSuperPlaylist();
        Assert.assertEquals(songFromSuperPlaylist, getSongTitle);


    }

    public void ValidateSongIsPlaying(){
        WebElement ValidateSongPlaying = driver.findElement(By.xpath("//button[@title='Click for a marvelous visualizer!']"));
        ValidateSongPlaying.click();
        Assert.assertTrue(ValidateSongPlaying.isDisplayed());
    }
    private void PlaySong() {
        WebElement PlaySong = driver.findElement(By.xpath("//span[@data-testid='play-btn']"));
        Actions actions = new Actions(driver);
        actions.click(PlaySong).perform();

    }


    private String getSongTitleFromSuperPlaylist() {

        WebElement superPlaylistSong = driver.findElement(By.xpath("//section[@id='playlistWrapper']//tr[@class='song-item']//td[@class='title']"));
        return superPlaylistSong.getText();
    }



    private void addToSuperPlaylist() {
        WebElement superPlaylist = driver.findElement(By.xpath("//*[starts-with(text(),'Flannel')]"));
        superPlaylist.click();
    }

    private String getSongTitleTxt() {
        WebElement getTitleTxt = driver.findElement(By.xpath("//section[@id='recentlyPlayedWrapper']//tr[@class='song-item']//td[@class='title']"));
        return getTitleTxt.getText();
    }

    private void clickAddToPlaylist() {
        WebElement addToPlaylist = driver.findElement(By.xpath("//button[contains(@title, 'Add selected songs')]"));
        addToPlaylist.click();
    }

    private void clickSongFromList() {
        WebElement songList = driver.findElement(By.xpath("//section[@id='recentlyPlayedWrapper']//tr[@class='song-item']"));
        songList.click();

    }

    private void clickViewAllBtn() {
        WebElement viewAllBtn = driver.findElement(By.xpath("//button[@data-testid='home-view-all-recently-played-btn']"));
        viewAllBtn.click();
    }
}
```

**Comment by Berichicko** [comment]
Hello

Homework  18 is cheeked! 

**I left some recommendations for the code.** 🔥

**Your grade: 80**  ✊


**Well, great job!** 🔥

💖Best wishes, the team HW (Alex)!💖

**Pull request merged**